### PR TITLE
Preserve links

### PR DESCRIPTION
**Pull request by mzpqnxow**
This supercedes PR #23 and resolives issues #22 and #21

I will close PR #23

### CODE REVIEW

**Comment by mzpqnxow** [comment]
I also added a pedantic improvement from @kmarty from his PR #16 which changes "../" to "${DEST}"

Since the incremental directory is always one directory up from the full directory, the existing code works fine, but in my opinion the more correct way to do this is as he suggested, by explicitly specifying the full destination root.

Sorry for lumping all of these commits into one branch, but I *think* they will not be controversial changes and so can all be accepted or rejected in whole

**Comment by mzpqnxow** [comment]
@cytopia I just noticed this in the documentation:

> By default the only rsync option used is --recursive.
This is due to the fact that some remote NAS implementations do not support
symlinks, changing owner, group or permissions (due to restrictive ACL's).
If you want to use any of those options you can simply append them.
See Example section for how to.

... so maybe you will want to close this PR, or remove the preservation of symbolic links and owners from it. To me, it seems that they should be backed up by default and disabled for the edge case (the NAS devices you mentioned)

**Comment by kmarty** [comment]
That "pedantic" improvement has a reason:
"../" way:
```
kmarty@Kmartys-MBP:~/workspace/linux-timemachine$ ./timemachine /tmp/src /tmp/dst --archive
kmarty@Kmartys-MBP:~/workspace/linux-timemachine$ ./timemachine /tmp/src /tmp/dst --archive
kmarty@Kmartys-MBP:~/workspace/linux-timemachine$ ./timemachine /tmp/src /tmp/dst --archive
kmarty@Kmartys-MBP:~/workspace/linux-timemachine$ du -sh /tmp/dst/*
7,0M    /tmp/dst/2019-02-14__17-01-46
7,0M    /tmp/dst/2019-02-14__17-01-48
7,0M    /tmp/dst/2019-02-14__17-01-50
  0B    /tmp/dst/current
```

Whereas "${DEST}/" way:
```
kmarty@Kmartys-MBP:~/workspace/linux-timemachine$ ./timemachine /tmp/src /tmp/dst --archive
kmarty@Kmartys-MBP:~/workspace/linux-timemachine$ ./timemachine /tmp/src /tmp/dst --archive
kmarty@Kmartys-MBP:~/workspace/linux-timemachine$ ./timemachine /tmp/src /tmp/dst --archive
kmarty@Kmartys-MBP:~/workspace/linux-timemachine$ du -sh /tmp/dst/*
7,0M    /tmp/dst/2019-02-14__17-03-17
  0B    /tmp/dst/2019-02-14__17-03-20
  0B    /tmp/dst/2019-02-14__17-03-22
  0B    /tmp/dst/current
```

The "../" doesn't work because "../current" doesn't have to be (and it isn't in this case) at "/tmp/dst/current".

File: timemachine [hunk 1]
```diff
@@ -196,15 +196,25 @@ if test -L "${DEST}/${BACKUP_LATEST}"; then
...
 		--delete \
 		--delete-excluded \
+		--times \
+		--perms \
+		--owner \
+		--group \
+		--links \
 		--partial-dir="${RSYNC_PARTIAL}" \
-		--link-dest="../${BACKUP_LATEST}" \
+		--link-dest="${DEST}/${BACKUP_LATEST}" \
```
Comment (cytopia) [inline timemachine]: This somehow breaks on Linux.

**Comment by cytopia** [comment]
@mzpqnxow as everything you PR'ed has already been implemented, except:

```diff
- --link-dest="../${BACKUP_LATEST}"
+ --link-dest="${DEST}/${BACKUP_LATEST}"
```
which was failing on Linux, I have reverted this specific change. This is still available in another other PR: https://github.com/cytopia/linux-timemachine/pull/29/

With this mentioned change I can go ahead and merge the documentation for cron.

**Pull request merged**